From 1fdab5e4c1f550300c92859c65b6349d50c847b4 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 25 Aug 2022 18:05:43 +0200
Subject: [PATCH 065/529] dt-bindings: interrupt-controller: stm32-exti: add
 property 'wakeup-parent'

Some EXTI interrupts can be routed to a parent controller able to
wakeup the SoC from suspend.

Add the optional property 'wakeup-parent' to EXTI binding to
specify the wakeup capable interrupt controller.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I8628a57dbc3dc939ba8a6fd52ab090a47c08cac1
---
 .../bindings/interrupt-controller/st,stm32-exti.yaml        | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/Documentation/devicetree/bindings/interrupt-controller/st,stm32-exti.yaml
+++ b/Documentation/devicetree/bindings/interrupt-controller/st,stm32-exti.yaml
@@ -42,6 +42,8 @@ properties:
     description:
       Interrupts references to primary interrupt controller
 
+  wakeup-parent: true
+
 patternProperties:
   '^exti-interrupt-map$':
     type: object
@@ -111,13 +113,15 @@ examples:
         interrupt-controller;
         #interrupt-cells = <2>;
         reg = <0x5000d000 0x400>;
+        wakeup-parent = <&pwr_irq>;
         exti-interrupt-map {
             #address-cells = <0>;
             #interrupt-cells = <2>;
             interrupt-map-mask = <0xffffffff 0>;
             interrupt-map =
                 <0 0 &intc GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>,
-                <3 0 &intc GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>;
+                <3 0 &intc GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>,
+                <55 0 &pwr_irq 0 IRQ_TYPE_EDGE_FALLING 0>;
         };
     };
 
